src/group_theory/submonoid/pointwise.lean

```diff
@@ -113,6 +112,63 @@ end submonoid
 
 namespace submonoid
 
+section monoid
+
+open_locale pointwise
+
+variables [comm_monoid G]
+
+/-- The submonoid of powers of a natural number of a commutative submonoid.
+
+This is available as an instance in the `pointwise` locale. -/
+protected def pointwise_has_pow : has_pow (submonoid G) ℕ :=
+{ pow := λ S n,
+  { carrier  := (S : set G) ^ n,
```

Comment:
Arghh, I've just had a horrible realization. There are two different definitions of the power of a set:

* `{2, 3} ^ 2 = {4, 9}`
* `{2, 3} ^ 2 = {2, 3} * {2, 3} = {4, 6, 9}`

Right now, you've implemented the first one for `submonoid` and the second one for `subgroup`!

@YaelDillies, any opinions here?

Reply:
As explained on Xena, both should be implemented, and the monoid/group power should have priority so that, if you happen to be working over `ℕ`, you can still spell `{2, 3} ^ 2 = {4, 9}` as `image (^2) {2, 3}` which is not awfully long compared to having to redefine `npow` inductively.